aea/helpers/env_vars.py

```diff
@@ -78,25 +80,44 @@ def replace_with_env_var(
 def apply_env_variables(
     data: Union[Dict, List[Dict]],
     env_variables: Mapping[str, Any],
+    path: Optional[List[str]] = None,
     default_value: Any = NotSet,
 ) -> JSON_TYPES:
     """Create new resulting dict with env variables applied."""
+    path = path or []
 
     if isinstance(data, (list, tuple)):
         result = []
         for i in data:
-            result.append(apply_env_variables(i, env_variables, default_value))
+            result.append(
+                apply_env_variables(
+                    data=i,
+                    env_variables=env_variables,
+                    path=path,
+                    default_value=default_value,
+                )
+            )
         return result
 
     if isinstance(data, MappingType):
         return {
-            apply_env_variables(k, env_variables, default_value): apply_env_variables(
-                v, env_variables, default_value
+            k: apply_env_variables(
+                data=v,
+                env_variables=env_variables,
+                path=[*path, k],
+                default_value=default_value,
             )
             for k, v in data.items()
         }
+
     if is_env_variable(data):
-        return replace_with_env_var(data, env_variables, default_value)
+        default_var_name = "_".join(path)
```

Comment:
We can auto generate the default environment variable using the json path for the variable.

Take this simple example of an override
```yaml
---
public_id: fetchai/http_server:0.22.0
type: connection
config:
  host: ${str:localhost}
  port: ${PORT:int:5000}
```

When loading value for `port`, `POST` will be used as the environment variable name. And since environment variable for `host` is not defined so it will be autogenerated as

`CONNECTION_HTTP_SERVER_CONFIG_HOST`

Reply:
1. there may be conflicting environmental variable names when combining components. Is there something we can do to prevent such, or minimize chances of running into such?
2. how does the developer know to set `CONNECTION_HTTP_SERVER_CONFIG_HOST`?

Reply:
1. The only way we can have conflicting env var names is to have two components with same name and same component types. That is possible when a component is published by the two different authors and pulled by a developer for a project. This seems highly unlikely to happen, but we can update the default var name generation to include author name at the beginning.

For example take the same component as above and assume that `http_server` was published by two different authors a and b as a connection. The default env var name for both should be `CONNECTION_HTTP_SERVER_CONFIG_HOST ` but that can be fixed if we append the author name at the beginning. eg `A_CONNECTION_HTTP_SERVER_CONFIG_HOST ` and `B_CONNECTION_HTTP_SERVER_CONFIG_HOST `. 

For now I suggest we keep this mechanism and I'll open a ticket for using author name as well for a future release

2. The docs, we have docs which explain how to use env vars and I'll extend them to explain auto generated env vars and the env var names can be user defined as well, that is not the default behaviour, the user can define the env var names as if they want to as explained above with the example of `PORT`.

Reply:
I agree that including the author at the beginning is safer.

Reply:
just an idea: we could potentially report which environmental variables are used in configurations to overwrite; which keys (not values) are present and which ones are not.